tests/interfaces/test_autograd.py

```diff
@@ -1111,3 +1112,30 @@ def test_multiple_hamiltonians_trainable(self, cost_fn, execute_kwargs, tol):
         res = np.hstack(qml.jacobian(cost_fn)(weights, coeffs1, coeffs2, dev=dev))
         expected = self.cost_fn_jacobian(weights, coeffs1, coeffs2)
         assert np.allclose(res, expected, atol=tol, rtol=0)
+
+
+class TestCustomJacobian:
+    def test_custom_jacobians(self):
+        class CustomJacobianDevice(DefaultQubit):
+            @classmethod
+            def capabilities(cls):
+                capabilities = super().capabilities()
+                capabilities["provides_jacobian"] = True
+                return capabilities
+
+            def jacobian(self, tape):
+                return np.array([1.0, 2.0, 3.0, 4.0])
+
+        dev = CustomJacobianDevice(wires=2)
+
+        @qml.qnode(dev, diff_method="device")
+        def circuit(v):
+            qml.RX(v, wires=0)
+            return qml.probs(wires=[0, 1])
+
+        d_circuit = qml.jacobian(circuit, argnum=0)
+
+        params = np.array(1.0, requires_grad=True)
+
+        d_out = d_circuit(params)
+        assert np.allclose(d_out, np.array([1.0, 2.0, 3.0, 4.0]))

```

Comment:
@antalszava I am trying to understand why this bug didn't cause a problem for us. What was `d_out` before this fix? Thanks very much!

Reply:
This is a very good question! Another good question is if this change would affect your custom device jacobians 🤔

Reply:
Now I see: We had actually worked around this by returning `np.array([gradient])` from our device's jacobian method, where gradient is itself a numpy array. I guess we just need to stop doing this starting with PL v0.23.0 then. Thanks for the heads up!